File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/lectures/adapter/LectureListSelectionAdapter.java
package de.tum.in.tumcampusapp.component.tumui.lectures.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.CheckBox;
import android.widget.CompoundButton;

import java.util.List;

import de.tum.in.tumcampusapp.R;
import de.tum.in.tumcampusapp.component.tumui.calendar.CalendarController;
import de.tum.in.tumcampusapp.component.tumui.calendar.model.CalendarItem;
import de.tum.in.tumcampusapp.utils.Utils;

public class LectureListSelectionAdapter extends BaseAdapter implements CompoundButton.OnCheckedChangeListener {
    private final int appWidgetId;
    private final CalendarController calendarController;
    private final List<CalendarItem> calendarItems;
    private final LayoutInflater mInflater;

    public LectureListSelectionAdapter(Context context, List<CalendarItem> cItems, int appWidgetId) {
        calendarItems = cItems;
        this.appWidgetId = appWidgetId;
        this.calendarController = new CalendarController(context);
        mInflater = LayoutInflater.from(context);
    }

    @Override
    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
        // save new preferences
        Utils.logv("Widget asked to change " + buttonView.getText()
                                                         .toString() + " to " + isChecked);
        if (isChecked) {
            calendarController.deleteLectureFromBlacklist(this.appWidgetId, (String) buttonView.getText());
        } else {
            calendarController.addLectureToBlacklist(this.appWidgetId, (String) buttonView.getText());
        }
    }

    @Override
    public int getCount() {
        return calendarItems.size();
    }

    @Override
    public Object getItem(int position) {
        return calendarItems.get(position);
    }

    @Override
    public long getItemId(int position) {
        return Long.parseLong(calendarItems.get(position)
                                           .getNr());
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View view = convertView;
        if (view == null) {
            view = mInflater.inflate(R.layout.list_timetable_configure_item, parent, false);
        }
        CheckBox checkBox = view.findViewById(R.id.timetable_configure_item);
        checkBox.setChecked(!calendarItems.get(position)
                                          .getBlacklisted());
        checkBox.setText(calendarItems.get(position)
                                      .getTitle());
        checkBox.setOnCheckedChangeListener(this);

        return view;
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/lectures/adapter/LectureListSelectionAdapter.kt
package de.tum.`in`.tumcampusapp.component.tumui.lectures.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.CheckBox
import android.widget.CompoundButton

import de.tum.`in`.tumcampusapp.R
import de.tum.`in`.tumcampusapp.component.tumui.calendar.CalendarController
import de.tum.`in`.tumcampusapp.component.tumui.calendar.model.CalendarItem
import de.tum.`in`.tumcampusapp.utils.Utils
import java.lang.Long.parseLong

class LectureListSelectionAdapter(
    context: Context,
    private val calendarItems: List<CalendarItem>,
    private val appWidgetId: Int
) : BaseAdapter(), CompoundButton.OnCheckedChangeListener {

    private val calendarController: CalendarController = CalendarController(context)
    private val inflater: LayoutInflater = LayoutInflater.from(context)

    override fun onCheckedChanged(buttonView: CompoundButton, isChecked: Boolean) {
        // save new preferences
        Utils.logVerbose("Widget asked to change ${buttonView.text} to $isChecked")
        if (isChecked) {
            calendarController.deleteLectureFromBlacklist(this.appWidgetId, buttonView.text as String)
        } else {
            calendarController.addLectureToBlacklist(this.appWidgetId, buttonView.text as String)
        }
    }

    override fun getCount() = calendarItems.size

    override fun getItem(position: Int) = calendarItems[position]

    override fun getItemId(position: Int) = parseLong(calendarItems[position].nr)

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view: View = convertView ?: inflater.inflate(R.layout.list_timetable_configure_item, parent, false)

        val checkBox = view.findViewById<CheckBox>(R.id.timetable_configure_item)
        checkBox.isChecked = !calendarItems[position].blacklisted
        checkBox.text = calendarItems[position].title
        checkBox.setOnCheckedChangeListener(this)

        return view
    }
}
